dispatcher/backend/src/routes/tasks/task.py

```diff
@@ -132,34 +186,32 @@ def post(self, task_id: str, token: AccessToken.Payload):
             logger.exception(exc)
             raise exc
 
+        task_id = task.id
         payload = {"worker": request_args["worker_name"]}
         try:
-            task_event_handler(task_id, TaskStatus.reserved, payload)
+            task_event_handler(
+                session=session,
+                task_id=task_id,
+                event=TaskStatus.reserved,
+                payload=payload,
+            )
         except Exception as exc:
             logger.exception(exc)
             logger.error("unable to create task. reverting.")
```

Comment:
Right! Sorry

Reply:
And `commit` is not performed if an exception forces the exit of the current @dbsession scope, a `rollback` is performed instead in case we were in a nested @dbsession (hence other stuff performed in a potential outer @dbsession scope could still be commited if the exception is catched in the outer code)